### ISSUE TITLE
Performance loss on PS Vita version.

### ISSUE THREAD

**Issue by Snowbro88** [libretro/uae4arm-libretro#23]
Hi, I know it's been a while but I would like to report that this core, at least in its version for PS Vita, has had a problem for several months, basically going from full speed in almost the entire catalog to running slow, like half the normal fps , the reason is unknown, maybe it was a change on the libretro side, I just wanted to comment on it in case that something can be done.

I tried it with the latest stable version of retroarch (v1.15.0) and replacing the .elf core with a version of this date from April 2, 2022 works normally, but I want it to work correctly again due to the different modifications that are made in the main interface of retroarch and I don't know if it generates any conflicts running an old core on a recent version and let's say it stays as a fully functional core.

Thank you.

**Comment by foxp4v** [comment]
Had the same issue, i'm using an old version of the core with a recent version of retroarch, seems to work by now.

**Comment by Chips-fr** [comment]
I reviewed the code and found some defect on audio side. See:

https://github.com/Chips-fr/uae4arm-rpi/issues/73#issuecomment-2496109019

I'm wondering if it could be the reason for the audio issue on vita, however i don't have one.
I can compile and share vpk file. Is there someone interested ?

**Comment by Snowbro88** [comment]
> I reviewed the code and found some defect on audio side. See:
> 
> [Chips-fr#73 (comment)](https://github.com/Chips-fr/uae4arm-rpi/issues/73#issuecomment-2496109019)
> 
> I'm wondering if it could be the reason for the audio issue on vita, however i don't have one. I can compile and share vpk file. Is there someone interested ?

Hi, I just received the notification to my email, and yes, of course I am interested in a vpk with the core fixed for PS Vita, I can test it in a few hours.

**Comment by Chips-fr** [comment]
You can download this file:

https://file.io/afTKQJkRirxg

There are 3 vpk inside:
- retroarch_vita_libretro_repo.vpk  : this one is based on this repo. It will helps me to understand if my compilation environment is working.
- retroarch_vita_chips_repo_libretro_branch.vpk : this one is based on my libretro branch. It's same as above with an added way to support whdload.
- retroarch_vita_chips_repo_libretro_audio_test_branch.vpk : Same as above with the audio update.

**Comment by Snowbro88** [comment]
@Chips-fr Understood, I can tell you the test results tonight or tomorrow.

By the way, do these vpks overwrite the official retroarch on vita or do they have a different game ID?

In that case, I should first uninstall the official version on my console.

Thanks for your interest.

**Comment by Chips-fr** [comment]
For the game ID i don't know.
They only contains uae4arm core.

**Comment by Snowbro88** [comment]
@Chips-fr 

Hi, I've already installed the 3 vpks but I couldn't make any comparison in their operation because when trying to load any game in any format, retroarch tells me there are no cores available.

I thought that this happened because I had a custom configuration of official retroarch but I deleted the configuration file so it's like a clean installation but this is not the problem, it seems that the applications can't find the core.

I also can't find the .self and .info of the core inside the vpks. Maybe it's possible to have those versions in .self and .info to manually add it to the version of retroarch that I have installed on my PS Vita and try this way.

Thanks for your interest.

**Comment by Chips-fr** [comment]
If i look in intermediate file in my compilation environment i can see the files you're talking about:

```
fred@fred-VirtualBox:~/libretro-super$ ls -al ./retroarch/pkg/vita/retroarch.vpk/vpk
total 6124
drwxrwxr-x 1 fred fred      82 nov.  30 09:05 .
drwxrwxr-x 1 fred fred       6 nov.  30 09:04 ..
-rw-rw-r-- 1 fred fred   88608 nov.  30 09:04 eboot.bin
drwxrwxr-x 1 fred fred      42 nov.  30 09:05 info
drwxrwxr-x 1 fred fred      52 nov.  30 09:04 sce_sys
-rw-rw-r-- 1 fred fred 6179063 nov.  30 09:05 uae4arm_libretro.self
fred@fred-VirtualBox:~/libretro-super$ ls -al ./retroarch/pkg/vita/retroarch.vpk/vpk/info
total 4
drwxrwxr-x 1 fred fred   42 nov.  30 09:05 .
drwxrwxr-x 1 fred fred   82 nov.  30 09:05 ..
-rw-rw-r-- 1 fred fred 1699 nov.  30 09:05 uae4arm_libretro.info
```


But if you look closely, eboot.bin is smaller here compared to what i have send to you. So perhaps things get merged together in eboot.bin.

Anyway if what i give you is not working, please find zip which contains uae4arm_libretro.self and .info for the 3 repo:

https://file.io/kEztu1z4helW

**Comment by Snowbro88** [comment]
@Chips-fr 

Hi again, now I was able to do the corresponding tests and this is what I can tell you:

retroarch_vita_libretro_repo.vpk : this version has the performance problem but it can run all adf, lha and hdf formats, it already did so for a while.

retroarch_vita_chips_repo_libretro_branch.vpk : this version also has performance problems and now also when running a game in lha format it causes a crash on PS Vita, a generic error c-128.

retroarch_vita_chips_repo_libretro_audio_test_branch.vpk : this version works perfectly in terms of speed, it seems to solve the performance and audio problem but it also has problems with the lha format.

About the lha format problem I have moved the core configuration in the Amiga A1200, A600 and auto types and in all of them it crashes, I can also notice that now there is the option of whdload type as files or hdfs and in both the lha format crashes, the hdf format no matter what whdload type is active, it works in both. In the libretro version there is no option of whdload type from core settings but both the lha and hdf formats work without problems.

In conclusion the third version seems to work correctly in performance it only has a problem when loading games in lha format, the adf and hdf formats work correctly.

I understand that you added compatibility with whdload formats (lha and hdf) but the core already had that compatibility previously, maybe that's why it now crashes with the lha format.

I hope I'm not abusing it but it would be possible to review the lha format issue, it is the best format from my point of view, fast loading times and lighter compared to hdfs

Thanks.

**Comment by Chips-fr** [comment]
I've made some add-on regarding whdload on my repo which works on raspberry but i feel like vita is more prone to regression, so it's better to add only the audio update.

I've made a pull request for the audio part for this repo:

https://github.com/libretro/uae4arm-libretro/pull/26

You can try following binary, It should be what will be the result after the above pull request:

https://transfert.free.fr/V36jFQ4

**Comment by Snowbro88** [comment]
@Chips-fr 

Hi, I've already tested your final version of UAE4ARM and now it works perfectly, good performance, correct audio and supports all formats, thank you very much for your work and time.

Just before closing this issue, I ask you if it is possible to make some more suggestions in a new issue to further improve this emu that performs really well on PS Vita.

Thanks and best regards.

**Comment by Chips-fr** [comment]
You can try on my repo or here but my time allocated to this project is quite low recently and i focus on improvements which interest me and are easy to do.

**Comment by Vecinagamer** [comment]
Hi @Chips-fr chips-fr  pleasePlease could you include the UAE4ARM core, which thanks to your great help works well, add it to the emu4vitaplus, it would be wonderful to have this core sincerely, thank you and congratulations for your great work.

**Comment by Snowbro88** [comment]
@Chips-fr Hi again boss, I left you one last request for the UAE4ARM core in your repository, I promise it will be the last for a while.

**Comment by Chips-fr** [comment]
It take time to the  pull request to be merged, if possible you can add comment in the pull request

**Comment by Snowbro88** [comment]
> It take time to the pull request to be merged, if possible you can add comment in the pull request

Hi, sorry for the delay in replying. I already commented to see if the Libretro admin can finally merge the fix into the PS Vita branch.

In any case, don't think the work was in vain. The Emu4vita++ developer has added this core to his project, and it works correctly here.

https://github.com/noword/Emu4VitaPlus/releases

The only thing that's still a bit missing is the crop overscan option to get rid of the black borders.

On the other hand, I found some games in LHA format that crash automatically, both licensed ones like Arabian Nights and unlicensed ones like Devils Temple. It seems to only affect that LHA format and only in specific cases. It seems that all games work correctly in ADF and HDF.

Well, thanks for your work, and let's hope it's finally added to Retroarch now that the new stable version is out.
